library/waku_thread/inter_thread_communication/requests/protocols/store_request.nim

```diff
@@ -14,31 +14,29 @@ import
 type StoreReqType* = enum
   REMOTE_QUERY ## to perform a query to another Store node
 
-type JsonStoreQueryRequest* = object
+type StoreRequest* = object
+  operation: StoreReqType
   jsonQuery: cstring
   peerAddr: cstring
   timeoutMs: cint
 
-type StoreRequest* = object
-  operation: StoreReqType
-  storeReq: pointer
-
 func fromJsonNode(
-    T: type JsonStoreQueryRequest, jsonContent: JsonNode
+    T: type StoreRequest, jsonContent: JsonNode
 ): Result[StoreQueryRequest, string] =
   let contentTopics = collect(newSeq):
     for cTopic in jsonContent["content_topics"].getElems():
       cTopic.getStr()
 
   let msgHashes = collect(newSeq):
-    for hashJsonObj in jsonContent["message_hashes"].getElems():
-      var hash: WakuMessageHash
-      var count: int = 0
-      for byteValue in hashJsonObj.getElems():
-        hash[count] = byteValue.getInt().byte
-        count.inc()
+    if jsonContent.contains("message_hashes"):
+      for hashJsonObj in jsonContent["message_hashes"].getElems():
+        var hash: WakuMessageHash
+        var count: int = 0
+        for byteValue in hashJsonObj.getElems():
+          hash[count] = byteValue.getInt().byte
+          count.inc()
 
-      hash
+        hash
```

Comment:
Same question as https://github.com/waku-org/waku-rust-bindings/pull/105#discussion_r1894249917

Reply:
> Same question as [waku-org/waku-rust-bindings#105 (comment)](https://github.com/waku-org/waku-rust-bindings/pull/105#discussion_r1894249917)

Yes good point! The reason for that is to make it work in the same way as relay: https://github.com/waku-org/nwaku/blob/09f05fefe2614d640277f96ae96ab4f2f12c8031/library/events/json_message_event.nim#L79